src/repo.rs

```diff
@@ -2100,6 +2101,40 @@ impl Repository {
         }
         Ok(ignored == 1)
     }
+
+    /// Look up the value of one git attribute for path. Equivalent to libgit2::git_attr_get.
+    pub fn get_attr(&self, flags: CheckAttributeFlags, path: &str, name: &str) -> Result<AttributeType, Error> {
```

Comment:
Could this perhaps be split into two APIs? One that returns an opaque `Attribute`, and then an API on `Attribute` to interpret what type it is?

Reply:
Thanks! 

Part of this was with the goal to make the API a bit safer (ie. representing the implicit enum of `git_attr_t` explicit). The C API returns something quite close to an enum, so it seems like an enum makes a lot of sense as a return value. 

Do you think it'd be better to keep it closer to the original API here? It would allow you to squeak out a bit more performance, perhaps?

Either way, I'll swap the to_string_lossy to a &[u8] for the value itself, and rename AttributeType to Attribute. 

I added some comments above as well, just general questions -- let me know if there's a better place to add that kind of stuff.

I'm following along from the header file [here](https://github.com/libgit2/libgit2/blob/master/include/git2/attr.h).

Reply:
Ah I'm not too too worried about performance here but more idioms and following the original documentation. Most of this library is written assuming you're also following along with the documentation of upstream libgit2 (as it's far more comprehensive), so having the same idioms and styles of APIs I think are important for doing the translation.

Reply:
I've been working on splitting the API up into two, but ran into a (potential?) safety problem.

The attribute pointer returned by the C implementation can only be read as byte string if the attribute is of type value. If we expose an `as_bytes` function on the opaque Attribute, we would first have to make the check to verify that it does indeed have the `git_attr_t` type of Value. 

The LibGit2 documentation is a bit unclear on the behaviour of trying to read a non-value attribute as a string. It seems undefined.

Does that seem reasonable? It defeats the purpose of splitting the API on a procedural level, although I'm happy to keep with this strategy, as it does make sure to stick with the documentation. 

Thanks for the help!

Reply:
Ah ok then yeah in that case we'd probably want it to return an `Option` and then it'd get checked  when the bytes are accessed